build.envd

```diff
@@ -0,0 +1,42 @@
+# syntax=v1
+# You can install envd with pip: `pip install envd`.
+# Make sure you already have docker installed.
+# To start the development of mosec, just run `envd up`, then you will get an
+# isolated and clean Python & Rust development environment.
+# To access GPU, you can run `envd up :gpu`. Make sure you have nvidia-docker
+# installed.
+
+
+def rust():
+    """Install Rust."""
+    install.apt_packages(name=["build-essential"])
+    run(
+        [
+            "curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y",
+        ]
+    )
+    runtime.environ(extra_path=["/home/envd/.cargo/bin"])
+
+
+def jax():
+    install.python_packages(
+        name=[
+            "jax[cuda] -f https://storage.googleapis.com/jax-releases/jax_cuda_releases.html",
+            "chex",
+        ]
+    )
+
+
+def build():
+    base(dev=True)
+    install.conda()
+    install.python()
+    install.python_packages(requirements="requirements/dev.txt")
+    rust()
+    # run(["make install"], mount_host=True)
+    runtime.init(["make install"])
```

Comment:
Ideally, we should use `run` instead of `runtime.init` since `run` is executed during build time while `runtime.init` is executed after the container is running. But I got an error here:

```
PermissionError: [Errno 13] Permission denied: 'mosec/_version.py'
```

<details>
    <summary>Details</summary>

```
 => /usr/bin/bash -c set -euo pipefail
curl --proto =https --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
                         13.8s
 => ERROR /usr/bin/bash -c set -euo pipefail
make install
                                                                             3.2s
------
 > importing cache manifest from docker.io/tensorchord/python-cache:envd-v0.3.8:
------
------
 > /usr/bin/bash -c set -euo pipefail
make install
:
#0 0.266 pip install -e .[dev,doc,plugin]
#0 0.616 Defaulting to user installation because normal site-packages is not writeable
#0 0.664 Obtaining file:///home/envd/mosec
#0 0.669   Installing build dependencies: started
#0 2.473   Installing build dependencies: finished with status 'done'
#0 2.475   Checking if build backend supports build_editable: started
#0 2.634   Checking if build backend supports build_editable: finished with status 'done'
#0 2.635   Getting requirements to build editable: started
#0 2.817   Getting requirements to build editable: finished with status 'done'
#0 2.820   Preparing editable metadata (pyproject.toml): started
#0 3.058   Preparing editable metadata (pyproject.toml): finished with status 'error'
#0 3.064   error: subprocess-exited-with-error
#0 3.064
#0 3.064   × Preparing editable metadata (pyproject.toml) did not run successfully.
#0 3.064   │ exit code: 1
#0 3.064   ╰─> [35 lines of output]
#0 3.064       Traceback (most recent call last):
#0 3.064         File "/opt/conda/envs/envd/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 351, in <module>
#0 3.064           main()
#0 3.064         File "/opt/conda/envs/envd/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 333, in main
#0 3.064           json_out['return_val'] = hook(**hook_input['kwargs'])
#0 3.064         File "/opt/conda/envs/envd/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 179, in prepare_metadata_for_build_editable
#0 3.064           return hook(metadata_directory, config_settings)
#0 3.064         File "/tmp/pip-build-env-nowo8j9n/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 451, in prepare_metadata_for_build_editable
#0 3.064           return self.prepare_metadata_for_build_wheel(
#0 3.064         File "/tmp/pip-build-env-nowo8j9n/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 377, in prepare_metadata_for_build_wheel
#0 3.064           self.run_setup()
#0 3.064         File "/tmp/pip-build-env-nowo8j9n/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 484, in run_setup
#0 3.064           super(_BuildMetaLegacyBackend,
#0 3.064         File "/tmp/pip-build-env-nowo8j9n/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 335, in run_setup
#0 3.064           exec(code, locals())
#0 3.064         File "<string>", line 71, in <module>
#0 3.064         File "/tmp/pip-build-env-nowo8j9n/overlay/lib/python3.9/site-packages/setuptools/__init__.py", line 108, in setup
#0 3.064           return distutils.core.setup(**attrs)
#0 3.064         File "/tmp/pip-build-env-nowo8j9n/overlay/lib/python3.9/site-packages/setuptools/_distutils/core.py", line 147, in setup
#0 3.064           _setup_distribution = dist = klass(attrs)
#0 3.064         File "/tmp/pip-build-env-nowo8j9n/overlay/lib/python3.9/site-packages/setuptools/dist.py", line 475, in __init__
#0 3.064           _Distribution.__init__(
#0 3.064         File "/tmp/pip-build-env-nowo8j9n/overlay/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 283, in __init__
#0 3.064           self.finalize_options()
#0 3.064         File "/tmp/pip-build-env-nowo8j9n/overlay/lib/python3.9/site-packages/setuptools/dist.py", line 904, in finalize_options
#0 3.064           ep(self)
#0 3.064         File "/tmp/pip-build-env-nowo8j9n/overlay/lib/python3.9/site-packages/setuptools_scm/integration.py", line 127, in infer_version
#0 3.064           _assign_version(dist, config)
#0 3.064         File "/tmp/pip-build-env-nowo8j9n/overlay/lib/python3.9/site-packages/setuptools_scm/integration.py", line 60, in _assign_version
#0 3.064           maybe_version = _get_version(config)
#0 3.064         File "/tmp/pip-build-env-nowo8j9n/overlay/lib/python3.9/site-packages/setuptools_scm/__init__.py", line 162, in _get_version
#0 3.064           dump_version(
#0 3.064         File "/tmp/pip-build-env-nowo8j9n/overlay/lib/python3.9/site-packages/setuptools_scm/__init__.py", line 77, in dump_version
#0 3.064           with open(target, "w") as fp:
#0 3.064       PermissionError: [Errno 13] Permission denied: 'mosec/_version.py'
#0 3.064       [end of output]
#0 3.064
#0 3.064   note: This error originates from a subprocess, and is likely not a problem with pip.
#0 3.067 error: metadata-generation-failed
#0 3.067
#0 3.067 × Encountered error while generating package metadata.
#0 3.067 ╰─> See above for output.
#0 3.067
#0 3.067 note: This is an issue with the package mentioned above, not pip.
#0 3.067 hint: See above for details.
#0 3.143 make: *** [Makefile:7: install] Error 1
------
error: failed to solve: process "/usr/bin/bash -c set -euo pipefail\nmake install\n" did not complete successfully: exit code: 2
```

</details>